java/src/jmri/jmrit/display/MemoryIcon.java

```diff
@@ -577,20 +577,12 @@ public void doMouseClicked(JmriMouseEvent e) {
     }
 
     protected void editMemoryValue() {
-        JTextField newMemory = new JTextField(20);
-        if (getMemory().getValue() != null) {
-            newMemory.setText(getMemory().getValue().toString());
-        }
-        Object[] options = {Bundle.getMessage("ButtonCancel"), Bundle.getMessage("ButtonOK"), newMemory};
-        int retval = JOptionPane.showOptionDialog(this,
-                "Edit Current Memory Value", namedMemory.getName(),
-                0, JOptionPane.INFORMATION_MESSAGE, null,
-                options, options[2]);
-
-        if (retval != 1) {
-            return;
-        }
-        setValue(newMemory.getText());
+    
+        String reval = JOptionPane.showInputDialog(this,
+                                     "Edit Current Memory Value",
+                                     getMemory().getValue().toString());
```

Comment:
@bobjacobsen 
You have a bug here. If the value is null, getValue().toString() will throw a NPE.

Reply:
Is toString() even necessary here?